exporter/awsemfexporter/metric_declaration.go

```diff
@@ -136,6 +176,44 @@ func (m *MetricDeclaration) ExtractDimensions(labels map[string]string) (dimensi
 	return
 }
 
+// Initialize LabelMatcher with default values and compile regex string.
+func (lm *LabelMatcher) Init() (err error) {
+	// Throw error if no label names are specified
+	if len(lm.LabelNames) == 0 {
+		return errors.New("label matcher must have at least one label name specified")
+	}
+	if len(lm.Separator) == 0 {
+		lm.Separator = ";"
+	}
+	if len(lm.Regex) == 0 {
+		lm.Regex = ".+"
+	}
+	lm.compiledRegex = regexp.MustCompile(lm.Regex)
+	return
+}
+
+// Returns true if given set of labels matches the LabelMatcher's rules.
+func (lm *LabelMatcher) Matches(labels map[string]string) bool {
+	concatenatedLabels := lm.getConcatenatedLabels(labels)
+	return lm.compiledRegex.MatchString(concatenatedLabels)
+}
+
+// Concatenate label values of matched labels using separator defined by the LabelMatcher's rules.
+func (lm *LabelMatcher) getConcatenatedLabels(labels map[string]string) string {
+	buf := new(bytes.Buffer)
+	isFirstLabel := true
+	for _, labelName := range lm.LabelNames {
+		if isFirstLabel {
+			isFirstLabel = false
+		} else {
+			buf.WriteString(lm.Separator)
+		}
+
+		buf.WriteString(labels[labelName])
```

Comment:
What if configured label doesn't exist in the metric's labelset?
Since our default regex for one label is `.+`, I think maybe we should call it `unmatch` as long as any one of the labels doesn't exist.

Reply:
I'm open on whether allow label value as `nil` when concatenating labels. We just need to be consistent. If we allow `nil`, then we don't set default regex for label to `.+`.

Reply:
I think we can allow `nil` and also make the regex value mandatory, so that we leave it up to the customer to decide how to handle missing values in their regex string.